File: src/main/org/nlogo/api/WorldDimensions.java
package org.nlogo.api ;

// use this class to wrap up dimensions to resize the world 
// using WorldResizer

public strictfp class WorldDimensions
{
	public int minPxcor ;
	public int maxPxcor ;
	public int minPycor ;
	public int maxPycor ;

	public WorldDimensions( int minx , int maxx , int miny , int maxy )
	{
		minPxcor = minx ;
		maxPxcor = maxx ;
		minPycor = miny ;
		maxPycor = maxy ;
	}

	public int width()
	{
		return maxPxcor - minPxcor + 1 ;
	}

	public int height()
	{
		return maxPycor - minPycor + 1 ;
	}
}

File: src/main/org/nlogo/api/WorldDimensions.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

case class WorldDimensions(
    var minPxcor: Int, var maxPxcor: Int,
    var minPycor: Int, var maxPycor: Int) {
  def width = maxPxcor - minPxcor + 1
  def height = maxPycor - minPycor + 1
}
object WorldDimensions {
  def square(max: Int) =
    WorldDimensions(-max, max, -max, max)
}

